docs/whichkey/reference.md

```diff
@@ -0,0 +1,281 @@
+
+Description:
+A command used primarily for accepting non-characters key in which-key menu and pass in `when` clause via vscode shortcut.
+
+:::note
+
+#### Non-character key
+
+The which-key menu relies QuickPick's input. Because it is a text input, we can not capture non-character keys like arrow keys, tab key, and keys with
+modifier like C-x. We can capture those keys using vscode shortcut with `whichkeyVisible` as when clause of the shortcut.
+See [Extra](extra#use-non-character-keys) page for detail on usage.
+:::
+
+:::note
+
+#### Pass-in when clause
+
+Since vscode doesn't allows extension to read context key-value ([vscode#10471](https://github.com/microsoft/vscode/issues/10471)),
+we will have to relied on `when` clause evaluation in vscode shortcut to pass the context to which-key.
+See [conditional bindings](./extra.md#when) for more details on the usage.
+:::
+
+### Describe Bindings
+
+Command: `whichkey.describeBindings`
+
+Description:
+This is similar to [helm-descbinds](https://github.com/emacs-helm/helm-descbinds) which can search and execute the binding ([vscode-which-key#12](https://github.com/VSpaceCode/vscode-which-key/issues/12)).
+Note that it will only shows described bindings menu for the currently displayed bindings (and its sub-bindings).
+
+### Show Transient
+
+Command: `whichkey.showTransient`
+
+Args: [`TransientMenuConfig`](#transientmenuconfig) | `string`
+
+Description: Show a transient with the definition in the argument. The definition can be either
+
+- object of [`TransientMenuConfig`](#transientmenuconfig)
+- a `string` referencing the location of in the configuration to get `TransientMenuConfig`
+
+### Repeat Recent
+
+Command: `whichkey.repeatRecent`
+
+Description:
+A command to show a which-key menu with key 1 to 9 to repeat the most recently executed command bindings.
+
+### Recent Most Recent
+
+Command: `whichkey.repeatMostRecent`
+
+Description:
+A command to repeat the command binding most recently executed on which-key.
+
+### Toggle Zen Mode
+
+Command: `whichkey.toggleZenMode`
+
+Description:
+A command to toggle zen mode for transient menu, which will show/hide all the menu items to save screen real estate.
+
+### Open File
+
+Command: `whichkey.openFile`
+
+Description:
+This is a command to get around issue where vscode doesn't provide a single cross platform command to open file.
+See [vscode-which-key#26](https://github.com/VSpaceCode/vscode-which-key/issues/26).
+
+## Context
+
+The extension will also set the following context when applicable so it can be used in `when` clause of vscode shortcuts.
+
+`whichkeyActive` is a boolean that will be set to `true` when whichkey is active including the time when it is excluding a command.
+This is rarely used and most of the use case can be replaced with `whichKeyVisible`
+
+`whichkeyVisible` is a boolean that will be set to `true` when whichkey menu is visible. Note that this will be `false` when transient menu is visible.
```

Comment:
```suggestion
- `whichkeyVisible` is a boolean that will be set to `true` when the whichkey menu is visible.
  Note that this will be `false` when a transient menu is visible.
```